lib/membership_pair.dart

```diff
@@ -0,0 +1,37 @@
+import 'web_rtc_answer.dart';
+import 'web_rtc_ice_candidate.dart';
+import 'web_rtc_offer.dart';
+
+class MembershipPair {
+  final WebRtcOffer webRtcOffer;
+  final WebRtcAnswer webRtcAnswer;
+  final List <WebRtcIceCandidate> webRtcIceCandidates;
+
+  const MembershipPair({ this.webRtcOffer, this.webRtcAnswer, this.webRtcIceCandidates });
+
+  MembershipPair.fromAttributes(Map<String, dynamic> data) : this(
+    webRtcOffer: WebRtcOffer.fromAttributes(data['web_rtc_offer']),
+    webRtcAnswer: WebRtcAnswer.fromAttributes(data['web_rtc_answer']),
+    webRtcIceCandidates: (data['web_rtc_ice_candidates'] as List<Map<String, dynamic>>).map(
+      (webRtcOfferAttributes) => WebRtcIceCandidate.fromAttributes(webRtcOfferAttributes)
+    )
+  );
+
+  MembershipPair.empty() : this(
+    webRtcOffer: null,
+    webRtcAnswer: null,
+    webRtcIceCandidates: []
+  );
+
+  MembershipPair copyWith({
```

Comment:
@samandmoore I think you said there's a library that generates this kind of behavior? but this was easy enough to write I just did it myself for now.